tests/test_lock.py

```diff
@@ -0,0 +1,282 @@
+import os
+import shutil
+import time
+
+import pandas as pd
+import pytest
+
+import audbackend
+import audeer
+import audformat.testing
+import audiofile
+
+import audb
+
+
+class SlowFileSystem(audbackend.FileSystem):
+    r"""Emulate a slow file system.
+
+    Introduces a short delay when getting a file from the backend.
+    This ensures that timeouts are reached in the tests.
+
+    """
+    def _get_file(self, *args):
+        time.sleep(1)
+        super()._get_file(*args)
+
+
+audbackend.register(
+    'slow-file-system',
+    SlowFileSystem,
+)
+
+
+os.environ['AUDB_CACHE_ROOT'] = pytest.CACHE_ROOT
+os.environ['AUDB_SHARED_CACHE_ROOT'] = pytest.SHARED_CACHE_ROOT
+audb.config.REPOSITORIES = [
+    audb.Repository(
+        name=pytest.REPOSITORY_NAME,
+        host=pytest.FILE_SYSTEM_HOST,
+        backend='slow-file-system',
+    ),
+]
+
+
+DB_NAME = f'test_lock-{pytest.ID}'
+DB_ROOT = os.path.join(pytest.ROOT, 'db')
+DB_VERSION = '1.0.0'
+
+
+def clear_root(root: str):
+    audeer.rmdir(root)
+
+
+@pytest.fixture(
+    scope='function',
+    autouse=True,
+)
+def fixture_remove_db_from_cache():
+    root = audeer.path(pytest.CACHE_ROOT, DB_NAME)
+    clear_root(root)
+
+
+@pytest.fixture(
+    scope='module',
+    autouse=True,
+)
+def fixture_publish_db():
+
+    clear_root(DB_ROOT)
+    clear_root(pytest.FILE_SYSTEM_HOST)
+
+    # create db
+
+    db = audformat.testing.create_db(minimal=True)
+    db.name = DB_NAME
+    db.schemes['scheme'] = audformat.Scheme()
+    audformat.testing.add_table(
+        db,
+        'table',
+        'filewise',
+        num_files=[0, 1, 2],
+    )
+    db.save(DB_ROOT)
+    audformat.testing.create_audio_files(db)
+
+    # publish 1.0.0
+
+    audb.publish(
+        DB_ROOT,
+        DB_VERSION,
+        pytest.PUBLISH_REPOSITORY,
```

Comment:
Is this also on the slow file system?

Reply:
Maybe it worked locally as you had the database manually published there as well?

Reply:
I don't think we need to use the slow file system when we publish to the repository, as we are only using a timeout when we load from the repository.

Reply:
But you need to put a database in the repo on the slow file system, otherwise it cannot load from there?

Reply:
`host=pytest.FILE_SYSTEM_HOST`

OK, you are using the same folder then for the fast and slow repo.
Should indeed work.

Reply:
I now added an output message inside the slow file system, but this message is not shown in the runner (locally it shows the message). That means that in the runner it is not using the slow file system at all. The question is - WHY???

Reply:
Interesting: when I run all tests locally and not only `test_lock.py` I also don't see the sleep message I added to the slow file system. We delete the database from the cache in every tests, but maybe it's reading it from a different cache folder?